### PR TITLE
ADD: BEND-52 Field features to models sprint2

### PR DESCRIPTION
**Pull request by octapf**
Add field features to models  sprint2

- add id to direction model
- add selfTaught field to user model
- crate currentSituation model
- add currentSituation field to user model
- fix type modality field to Enum

### CODE REVIEW

**Review by PatyVilas — DISMISSED** [review]

**Comment by octapf** [comment]
@whitehatdevv please review

File: src/models/models.user.ts [hunk 1]
```diff
@@ -113,6 +114,17 @@ const UserSchema = new Schema({
...
+  selfTaught: {
+    type: Boolean,
+    required: true,
+    validate: {
+      validator: function (): Boolean {
+        return true;
+      },
+      message: props => `${props.value} is not a valid option`,
+    },
+  },
```
Comment (whitehatdevv) [inline src/models/models.user.ts:126]: It's not needed a validator here, obiusly a boolean can only have true/false cases.

**Review by whitehatdevv — CHANGES_REQUESTED** [review]

**Comment by whitehatdevv** [comment]
Also @PatyVilas where is the REGEX to validate the new fields?

File: src/models/models.user.ts (hunk 1 above)
Comment (octapf) [reply src/models/models.user.ts:126]: Ok i will erase it

**Review by PatyVilas — DISMISSED** [review]

**Review by Kehomaxd — APPROVED** [review]

**Review by davidcode10 — APPROVED** [review]
approved

**Review by PatyVilas — APPROVED** [review]

**Review by OvidioSantoro — APPROVED** [review]

**Pull request merged**